src/Containers/PanelCard/PanelCard.tsx

```diff
@@ -1,28 +1,123 @@
-import React from 'react';
-import { Button } from '@Inputs/Button/Button';
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+import { StyledIcon } from 'styled-icons/types';
+import { CheckCircle } from '@styled-icons/fa-solid/CheckCircle';
+import { TimesCircle } from '@styled-icons/fa-solid/TimesCircle';
+import { MainTheme } from '@Themes';
+import { flex } from '@Utils/Mixins';
 import { CardProps, Card } from '../Card/Card';
+import { Loading as L } from '../Loading/Loading';
+
+export enum OperationState {
+    isFailure,
+    isSuccess,
+    isLoading,
+    isUnknown,
+}
 
 export interface IPanelCardProps extends CardProps {
-    /** callback for cancel loading file */
-    onCancelLoading?: () => void;
+    /** name of the file */
+    name?: string;
     isFailureMessage?: string;
     isSuccessMessage?: string;
     isLoadingMessage?: string;
-    cancelLoadingText?: string;
+    operationState?: OperationState;
+    isSuccessIcon?: StyledIcon;
+    isFailureIcon?: StyledIcon;
+    isSuccessIconColor?:string;
+    isFailureIconColor?:string;
```

Comment:
I tried and it would be great if you do not use the default capability of all the options. say you have isSuccessOptions?:{
icon:{
type:StyledIcon;
color:string;
height:number;
}
message:string;
button:React.ReactElement;
}

now say you have a story where isSuccessOptions={
button:<Button>Dismiss</Button>
}

you lose all the defaults for the rest of is success options.